demos/get_fractal_demos.sh

```diff
@@ -1,17 +1,13 @@
-git clone https://github.com/fractal-analytics-platform/fractal-demos.git
-
 source config.env
 
-echo "FRACTAL_DEMOS_GIT=$FRACTAL_DEMOS_GIT"
-if [ ! -z "${FRACTAL_DEMOS_GIT}" ]; then
-    cd fractal-demos
-    git checkout $FRACTAL_DEMOS_GIT
-    
-    EXIT_CODE=$?
-    if [ $EXIT_CODE -ne 0 ]; then
-        check_exit_code "Checking out to fractal-demos@$FRACTAL_DEMOS_GIT (EXIT_CODE=$EXIT_CODE)"
-        exit 1
-    fi
-    
-    cd ..
+if [ -z "${FRACTAL_DEMOS_GIT}" ]; then
+    FRACTAL_DEMOS_GIT=main
+fi
+
+git clone --single-branch --branch $FRACTAL_DEMOS_GIT https://github.com/fractal-analytics-platform/fractal-demos.git
```

Comment:
Great, please use it consistently also for other git-clone commands

Reply:
> please use it consistently also for other git-clone commands

I take it back. It cannot be used in the tasks case, because we cannot set a default value to `main`.
Maybe we should stick with the naive approach (git clone and git checkout)? That one works in all known cases.